llarp/service/context.cpp

```diff
@@ -96,20 +96,21 @@ namespace llarp
           ++itr;
         }
       }
-
-      auto ep = getFirstEndpoint();
-      if(!ep)
-        return;
       std::vector< RouterID > expired;
       m_Router->nodedb()->visit([&](const RouterContact &rc) -> bool {
         if(rc.IsExpired(now))
           expired.emplace_back(rc.pubkey);
         return true;
       });
-      // TODO: we need to stop looking up service nodes that are gone forever
-      // how do?
-      for(const auto &k : expired)
-        ep->LookupRouterAnon(k);
+      ForEachService([&](const std::string &,
+                         const std::shared_ptr< Endpoint > &ep) -> bool {
+        // TODO: we need to stop looking up service nodes that are gone forever
+        // how do?
+        for(const auto &k : expired)
+          if(!ep->LookupRouterAnon(k))
+            return false;
```

Comment:
Can I just say I really don't like this bracing style (or the other one we have), I'd prefer:

```c++
for(const auto &k : expired) {
  if(!ep->LookupRouterAnon(k)) {
    return false;
  }
}
```

but I don't care that much.